mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -77,6 +77,20 @@ class Word2Vec extends Serializable with Logging {
   private var numIterations = 1
   private var seed = Utils.random.nextLong()
   private var minCount = 5
+  private var maxSentenceLength = 1000
+
+  /**
+   * sets the maxSentenceLength, maxSentenceLength is used as the threshold for cutting sentence
+   * into chunks when it is too long. (default: 1000)
+   * @param maxSentenceLength the maxSentenceLength allowed.
+   *                          recommend to set it large enough to respect reasonable long sentences
+   *                          while not overflow memory
+   * @return this object
+   */
+  def setMaxSentenceLength(maxSentenceLength: Int): this.type = {
```

Comment:
I think we should target `1.6.1` here: `@Since("1.6.1")` - overall I'd view this PR as a bugfix (though adding the parameter is a minor extra feature). I think we'd want to include this in branch-1.6, and possibly even think about backporting the core changes to branch-1.5